extension-style/src/main/java/com/mapbox/maps/extension/style/sources/generated/GeoJsonSource.kt

```diff
@@ -1116,9 +1207,39 @@ class GeoJsonSource(builder: Builder) : Source(builder.sourceId) {
 }
 
 /**
- * DSL function for [GeoJsonSource].
+ * DSL function for [GeoJsonSource] performing parsing using background thread.
+ * Immediately returns [GeoJsonSource] with no data set and starts preparing actual data
+ * using a worker thread.
+ *
+ * If using runtime styling:
+ *
+ * loadStyle(style(Style.DARK) {
+ *   +geoJsonSource(id) {
+ *    featureCollection(collection)
+ *   }
+ *   ...
+ * }
+ *
+ * compositing style will be performed correctly under the hood.
+ * [Style.OnStyleLoaded] will be emitted in correct moment of time when all sources are parsed.
```

Comment:
these only applies to using the StyleExtension object to load the style. If uses just use this single method, we can not guarantee the [Style.OnStyleLoaded] will be emitted in correct moment of time.

We could explain the behaviour like: 
Immediately returns [GeoJsonSource] with no data set and starts preparing actual data using a worker thread, and call the `setData` after the parsing is finished.

And this sync processing also doesn't apply to geoJson urls, it only applies to the Feature, FeatureCollection and  Geometry types.

Reply:
Updated docs to state it clearer.